src/main/java/org/dataconservancy/packaging/tool/cli/RdfContentProvider.java

```diff
@@ -0,0 +1,223 @@
+
+    /*
+     * TODO - Update doc throughout this method
+     * We manually build the IPM tree here. Fundamentally, we're doing three things:
+     * 1) Creating "directory" nodes that correspond to a domain object.
+     * 2) Creating "content" nodes that correspond to a domain object that
+     *    describes associated content.
+     * 3) Arranging these nodes into a tree structure of our liking.
+     */
+    public Node getIpmModel() {
+        // Must use an array to allow root node to be set inside lambda function below.
+        final Node[] root = new Node[1];
+
+        String msgFmt = "Creating %s IPM node named '%s' for domain object %s";
+
+        // For each subject resource that is not anonymous, create a node.
+        // - If the node type is an Image or File, then it will be a content node.
+        // - Otherwise, make a directory node.
+        // It is assumed that all folders will precede their child nodes/folders.
+        domainObjects.listSubjects().forEachRemaining(subject -> {
+            if (subject.isAnon()) {
+                LOG.debug("Skipping IPM node creation for anonymous resource '{}'", subject.getId().toString());
+                return;
+            }
+
+            URI u = URI.create(subject.getURI());
+
+            // Hash URIs do not get their own node; they will be considered to be a single node.
+            // TODO - Is this needed for JenaModel input? (copied from buildModelTree)
+            if (u.getFragment() != null) {
+                LOG.debug("Skipping IPM node creation for hash URI resource '{}'", subject.getURI());
+                return;
+            }
+
+            // What type of node is this?  What is its name?
+            String type  = subject.getProperty(hasType ).getObject().toString();
+            String title = subject.getProperty(hasTitle).getObject().toString();
+            boolean isImage  = type.equals("http://dataconservancy.org/business-object-model#Metadata");
+            boolean isFile   = type.equals("http://dataconservancy.org/business-object-model#File");
+            boolean isFolder = type.equals("http://dataconservancy.org/business-object-model#DataItem");
+
+            // Create the new package tree node and insert it into the tree.
+            Node n = new Node(u);
+            n.setDomainObject(u);
+
+            // Assign node file info and parent depending on node type.
+            // Note that file info must be assigned before finding/assigning parent.
+            if (isImage || isFile) {
+                LOG.info(String.format(msgFmt, "file", title, subject.getURI()));
+                Property prop = isImage ? metadataFor : isMemberOf;
+                String parentID = subject.getProperty(prop).getObject().toString();
+                assignFileInfo(title, parentID, contentPath, domainObjects, n, true);
+                assignParentNode(n, root[0]);
```

Comment:
Where is `root[0]` ever assigned?  You'll get an NPE if `assignParentNode` is called, right?  Is there test coverage for this block?